activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -316,12 +316,16 @@ def exists?(conditions = :none)
         MSG
       end
 
+      exists_some?(conditions: conditions)
+    end
+
+    def exists_some?(conditions: :none, min: 1, max: nil)
```

Comment:
No, it's not possible. `min` and `max` are necessary further. I'd try to unfold the underyling logic:

1. `relations.exists?`: `min = 1`, `max = nil` (`nil` means *no bound*). So the condition would be something like:
```ruby
relation.limit(1).pluck(1).length == 1
```
2. `relations.many?`: `min = 2`, `max = nil`. So the condition would be something like:
```ruby
relation.limit(2).pluck(1).length == 2
```
3. `relations.one?`: `min = 1`, `max = 1`. So the condition would be something like:
```ruby
relation.limit(2).pluck(1).length == 1
```

So the above logic could be generalized by:

```ruby
size = relation.limit(max ? max + 1 : min).pluck(1).length
size >= min && (!max || size <= max)
```

Reply:
So, it won't be possible to build `size >= min && size <= max` condition knowing only `limit`.

Reply:
It is possible, it just doesn't make sense given `#exists_some?`. If you renamed the method to `#exists_count` and moved the count comparisons to `#one?` and `#many?` respectively, it would cut down on some of the complexity. That was just an idea though, both approaches are valid.